### PR TITLE
[dataset] change padding value to 0

### PR DESCRIPTION
**Pull request by robin1001**

### CODE REVIEW

**Comment by xingchensong** [comment]
和blankid(0)冲突了？

**Comment by robin1001** [comment]
If we pad -1, then we have to change it to 0 or other values if positive int is requried. 0 is simpler for downstream task.

**Comment by robin1001** [comment]
image

已验证对 CTC 模块无影响，CTC loss 计算中无论 padding value 是多少，是 0，负值，h还是其他值都会忽略。

**Comment by robin1001** [comment]
``` python
import torch
import torch.nn.functional as F
from torch import nn

# Target are to be padded
T = 50      # Input sequence length
C = 20      # Number of classes (including blank)
N = 2       # Batch size
S = 10      # Target sequence length of longest target in batch (padding length)
S_min = 5  # Minimum target length, for demonstration purposes
# Initialize random batch of input vectors, for *size = (T,N,C)
input = torch.randn(T, N, C).log_softmax(2).detach().requires_grad_()
# Initialize random batch of targets (0 = blank, 1:C = classes)
target = torch.randint(low=1, high=C, size=(N, S), dtype=torch.long)
input_lengths = torch.full(size=(N,), fill_value=T, dtype=torch.long)
target_lengths = torch.randint(low=S_min, high=S, size=(N,), dtype=torch.long)
print(target_lengths)
ctc_loss = nn.CTCLoss()

loss = ctc_loss(input, target, input_lengths, target_lengths)
print('rand', loss)

for i in range(N):
    target[i, target_lengths[i]:] = 0
loss = ctc_loss(input, target, input_lengths, target_lengths)
print('fill 0', loss)

for i in range(N):
    target[i, target_lengths[i]:] = -1
loss = ctc_loss(input, target, input_lengths, target_lengths)
print('fill -1', loss)
```

附测试代码。

**Comment by Mddct** [comment]
pad value应该和ignore id 保持一致

**Comment by robin1001** [comment]
ignore id 用到 decoder loss 的计算中，目前现有模块会重新根据 label lengths 在后续处理中把 pad value 修改成 ignore id。

**Review by xingchensong — APPROVED** [review]
